### PR TITLE
tools/serial.inc.mk: Support miniterm.py.

### PR DESCRIPTION
**Pull request by jcarrano**
### Contribution description

miniterm.py is a simple terminal program that is included with pyserial. This means that it is available wherever pyterm can work. It allows raw access, does line translation and passes through special characters.

For test scripts, a terminal that does not modify the input and output streams, configured without local echo, is preferred as it ensures the test setup is introducing as few noise as possible.

### Testing procedure

With this change, `tests/shell` uses miniterm.py. Just run the tests. Use a real board, native has no real TTY.

```
$ cd tests/shell
$ BOARD=samr21-xpro make all
$ BOARD=samr21-xpro make flash
$ BOARD=samr21-xpro make test
```

### Issues/PRs references

This PR is part of  #10994 and it's purpose is to ease shell automation.
Split from  #10788 .

### CODE REVIEW

File: tests/shell/Makefile [hunk 1]
```diff
@@ -1,4 +1,5 @@
 DEVELHELP=0
+
```
Comment (aabadie) [inline tests/shell/Makefile]: unrelated

File: tests/shell/Makefile [hunk 2]
```diff
@@ -12,4 +13,8 @@ BOARD_BLACKLIST += chronos
 
 TEST_ON_CI_WHITELIST += all
 
+ifneq ($(BOARD), native)
+	RIOT_TERMINAL=miniterm.py
```
Comment (aabadie) [inline tests/shell/Makefile]: Looks like there's tab here. The indent should be 2 spaces.

File: makefiles/tools/serial.inc.mk [hunk 3]
```diff
@@ -19,4 +19,7 @@ ifeq ($(RIOT_TERMINAL),pyterm)
 else ifeq ($(RIOT_TERMINAL),picocom)
     export TERMPROG  ?= picocom
     export TERMFLAGS ?= --nolock --imap lfcrlf --echo --baud "$(BAUD)" "$(PORT)"
+else ifeq ($(RIOT_TERMINAL),miniterm.py)
+    export TERMPROG  ?= miniterm.py
+    export TERMFLAGS ?= --eol CRLF "$(PORT)" "$(BAUD)"
```
Comment (aabadie) [inline makefiles/tools/serial.inc.mk]: Maybe also enable `--echo` ? It's the kind of useful things when using the shell.

Comment (jcarrano) [reply makefiles/tools/serial.inc.mk]: With local echo you end up having _double_ echo, as the shell echoes too, and miniterm does not do line buffering, eg.:

```
> hheellpp
```

File: tests/shell/Makefile [hunk 4]
```diff
@@ -12,4 +12,10 @@ BOARD_BLACKLIST += chronos
 
 TEST_ON_CI_WHITELIST += all
 
+# In order to properly test and debug the shell it is better to use a terminal
```
Comment (aabadie) [inline tests/shell/Makefile]: Why just set this in `tests/shell` only ? I would prefer to have this set globally in `Makefile.tests_common`.

**Comment by jcarrano** [comment]
@aabadie I rebased this on top of your indentation fix.

**Comment by cladmi** [comment]
When testing https://github.com/RIOT-OS/RIOT/pull/10952 I found that having local echo may even be hurting for testing as it introduces more text in the tested text.

**Comment by cladmi** [comment]
I would prefer to move changing the test configuration in a different PR as it relates to the main tracking issue and not to adding the support.

Adding support should be done alone and it does not hurt anything.
Changing the test configuration is a different thing that may have consequences.

File: makefiles/tools/serial.inc.mk (hunk 3 above)
Comment (cladmi) [reply makefiles/tools/serial.inc.mk]: Having local echo messes with the tests also as the sent string is matched by pexpect.

File: tests/shell/Makefile [hunk 5]
```diff
@@ -12,4 +12,10 @@ BOARD_BLACKLIST += chronos
...
+# program that does not modify the input and output streams and has no buffering.
+ifneq ($(BOARD), native)
+  RIOT_TERMINAL?=miniterm.py
```
Comment (smlng) [inline tests/shell/Makefile]: please put spaces around `?=` look nicer and (more) inline with the rest of our makefiles stuff.

Comment (jcarrano) [reply tests/shell/Makefile]:
```suggestion
  RIOT_TERMINAL ?= miniterm.py
```

**Comment by jcarrano** [comment]
#11034 makes this PR irrelevant. Socat is a better tool. I'm closing this.
